File: backend/utils/response.py
def success(data=None, message="요청이 성공적으로 처리되었습니다."):
    return {
        "success": True,
        "data": data or {},
        "message": message,
        "error_code": None,
    }


def failure(message="요청 처리에 실패했습니다.", error_code="UNKNOWN_ERROR", data=None):
    return {
        "success": False,
        "data": data or {},
        "message": message,
        "error_code": error_code,
    }

File: frontend/src/utils/response.ts
// utils/response.ts

export const extractErrorMessage = (resData: any): string => {
  if (!resData) return '에러 응답이 없습니다.';

  return (
    resData.message ||
    resData.data?.non_field_errors?.[0] ||
    resData.data?.detail ||
    resData.error_code ||
    '알 수 없는 오류가 발생했습니다.'
  );
};
